src/main/java/bio/terra/common/flagsmith/FlagsmithClient.java

```diff
@@ -0,0 +1,101 @@
+package bio.terra.common.flagsmith;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import bio.terra.common.exception.BadRequestException;
+import com.flagsmith.exceptions.FlagsmithClientError;
+import com.flagsmith.models.Flags;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+
+public class FlagsmithClient {
+  private static final Logger LOGGER = getLogger(FlagsmithClient.class);
+  private final com.flagsmith.FlagsmithClient flagsmith;
+
+  /**
+   * Create a new flagsmith client
+   *
+   * @param flagsmithApiUrl API Url for flagsmith service deployment
+   * @param flagsmithClientSideKey flagsmith client side key
+   */
+  public FlagsmithClient(String flagsmithApiUrl, String flagsmithClientSideKey) {
+    if (StringUtils.isEmpty(flagsmithApiUrl) || StringUtils.isEmpty(flagsmithClientSideKey)) {
+      throw new BadRequestException("flagsmithApiUrl and flagsmithClientSideKey must be non-empty");
+    }
+
+    this.flagsmith =
+        com.flagsmith.FlagsmithClient.newBuilder()
+            .withApiUrl(flagsmithApiUrl)
+            .setApiKey(flagsmithClientSideKey)
+            .build();
+  }
+
+  /**
+   * Check if a feature is enabled
+   *
+   * @param feature {@link Features}
+   * @return true if the feature is enabled
+   */
+  public boolean isFeatureEnabled(Features feature) {
+    return isFeatureEnabled(feature.featureName, /*userEmail=*/ null).orElse(false);
+  }
+
+  /**
+   * Check if a feature is enabled
+   *
+   * @param feature {@link Features}
+   * @param userEmail user email to check for
+   * @return true if the feature is enabled for the user email
+   */
+  public boolean isFeatureEnabled(Features feature, @Nullable String userEmail) {
+    return isFeatureEnabled(feature.featureName, userEmail).orElse(false);
+  }
+
+  /**
+   * Check if a feature is enabled
+   *
+   * @param feature the name of the feature
+   * @param userEmail user email to check for
+   * @return {@code Optional.of(true)} if the feature is enabled for user email; {@code
+   *     Optional.of(false)} if the feature is disabled for user email; {@code Optional.empty()} if
+   *     the feature does not exist
+   */
+  public Optional<Boolean> isFeatureEnabled(String feature, @Nullable String userEmail) {
+    try {
+      return Optional.of(getFlags(userEmail).isFeatureEnabled(feature));
+    } catch (Exception e) {
+      LOGGER.debug("failed to fetch feature flag value", e);
+      return Optional.empty();
+    }
+  }
+
+  private Flags getFlags(String userEmail) throws FlagsmithClientError {
+    if (userEmail != null) {
+      flagsmith.getIdentityFlags(userEmail, Map.of("email_address", userEmail));
+    }
+    return flagsmith.getEnvironmentFlags();
+  }
+
+  // list of features
+  public enum Features {
```

Comment:
It seems pretty painful that we need to maintain the set of features in TCL. Are there any drawbacks to letting consumers manage the subset of flags they need and pass in string?

Reply:
+1 to this. 
Agree that it is an overhead. I debated it myself and here is the reason i stuck to it - 

- consolidating the list of features here: we dont need to hard code it in multiple repos. lettng consumers pass it as a param: hard coded values to be added in multiple repos, and is the value is incorrect a default 'false' is returned
- ideally this is done only once per major project/feature: for example , add it only once for AWS. hence the overhead is minimal
- With monorepo above overhead will further be reduced 

Not strongly opinionated on using enum though

Reply:
removed since the service code is changed to be used in CLI.